### PR TITLE
net-dns/doggo: new ebuild, add 1.0.4

### PR DESCRIPTION
**Pull request by hhoffstaette**
dig/host now require a full bind installation, so doggo can act as standalone replacement.
It has many nice features too.

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: net-dns/doggo/doggo-1.0.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+src_compile() {
+	emake build-cli VERSION=${PV} || die
+}
+
+src_install() {
+	newbin bin/${PN}.bin ${PN}
+
+	for comp in bash fish zsh; do
+		bin/${PN}.bin completions $comp > "${WORKDIR}"/${PN}.$comp
```
Comment (thesamesam) [inline net-dns/doggo/doggo-1.0.4.ebuild:28]: Missing `die`.

File: net-dns/doggo/doggo-1.0.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+BDEPEND="dev-lang/go"
+
+src_compile() {
+	emake build-cli VERSION=${PV} || die
+}
+
+src_install() {
+	newbin bin/${PN}.bin ${PN}
+
+	for comp in bash fish zsh; do
```
Comment (thesamesam) [inline net-dns/doggo/doggo-1.0.4.ebuild:27]:
```suggestion
	local comp
	for comp in bash fish zsh; do
```

File: net-dns/doggo/doggo-1.0.4.ebuild [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+HOMEPAGE="https://github.com/mr-karan/doggo"
+
+SRC_URI="https://github.com/mr-karan/${PN}/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz
+	https://www.applied-asynchrony.com/distfiles/${P}-deps.tar.xz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64"
+
+BDEPEND="dev-lang/go"
```
Comment (thesamesam) [inline net-dns/doggo/doggo-1.0.4.ebuild:18]: The eclass sets this.

File: net-dns/doggo/doggo-1.0.4.ebuild [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+src_compile() {
+	emake build-cli VERSION=${PV} || die
```
Comment (thesamesam) [inline net-dns/doggo/doggo-1.0.4.ebuild:21]: `emake` dies for us.

File: net-dns/doggo/doggo-1.0.4.ebuild [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+EAPI=8
+
+inherit go-module shell-completion
+
+DESCRIPTION="Command-line DNS Client for Humans"
+HOMEPAGE="https://github.com/mr-karan/doggo"
+
+SRC_URI="https://github.com/mr-karan/${PN}/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz
+	https://www.applied-asynchrony.com/distfiles/${P}-deps.tar.xz"
```
Comment (thesamesam) [inline net-dns/doggo/doggo-1.0.4.ebuild:12]: Please consider using my [script](https://github.com/thesamesam/sam-gentoo-scripts/blob/main/maint/bump-go) or something similar so you get a smaller and reproducible tarball.

Comment (hhoffstaette) [reply net-dns/doggo/doggo-1.0.4.ebuild:12]: I spent half an hour trying to make bump-go work, and the created tarball is exactly the same size as the one created by dev-go/go-dep-tarball (which I used). In other words this isn't getting any smaller.

File: net-dns/doggo/doggo-1.0.4.ebuild (hunk 1 above)
Comment (hhoffstaette) [reply net-dns/doggo/doggo-1.0.4.ebuild]: done

File: net-dns/doggo/doggo-1.0.4.ebuild (hunk 2 above)
Comment (hhoffstaette) [reply net-dns/doggo/doggo-1.0.4.ebuild:26]: done

File: net-dns/doggo/doggo-1.0.4.ebuild (hunk 4 above)
Comment (hhoffstaette) [reply net-dns/doggo/doggo-1.0.4.ebuild]: done

File: net-dns/doggo/doggo-1.0.4.ebuild (hunk 3 above)
Comment (hhoffstaette) [reply net-dns/doggo/doggo-1.0.4.ebuild]: done

**Pull request closed without merging**